src/System.Private.CoreLib/src/System/Utf8String.Manipulation.cs

```diff
@@ -76,29 +122,620 @@ public Utf8String Substring(int startIndex, int length)
             return InternalSubstring(startIndex, length);
         }
 
-        // Slice intended to be used by the compiler only to provide indexer with range parameter functionality.
-        // Developers should be using Substring method instead.
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [ComponentModel.EditorBrowsable(ComponentModel.EditorBrowsableState.Never)]
-        public Utf8String Slice(int startIndex, int length) => Substring(startIndex, length);
+        public SplitResult Split(char separator, Utf8StringSplitOptions options = Utf8StringSplitOptions.None)
+        {
+            if (!Rune.TryCreate(separator, out Rune rune))
+            {
+                throw new ArgumentOutOfRangeException(
```

Comment:
Throwhelper or internal throwing-method as in other places?
(same below)

Reply:
They originally did use `ThrowHelper`, but it looks like that's potentially discouraged outside of very hot paths. See https://github.com/dotnet/coreclr/pull/26711#discussion_r325450155.